### PR TITLE
Bugfix.docs edits development directory

### PR DESCRIPTION
**Pull request by selhorn**
Updated conf.py to remove reference to tests so that rst files including test in the name would build
Fixed style issues that were preventing code from being displayed
Fixed passive voice issues
Removed extra issues file from dev directory and put the info into usage\filing-issues file
First round of edits to make the content conform more closely to F5 style

### CODE REVIEW

File: docs/development/code-conventions.rst [hunk 1]
```diff
@@ -1,32 +1,18 @@
...
-When writing your modules and their accompanying tests and docs, please
-follow the below coding conventions.
+When you write modules and their accompanying tests and docs, follow these coding conventions.
 
 Use the complex/structure map format
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+------------------------------------
 
-In reference to Jeff Geerling's page `here`_, this format looks like this.
+In reference to Jeff Geerling's page, this format looks like this.
```
Comment (caphrim007) [inline docs/development/code-conventions.rst]: This line contains a URL ref, this shouldn't be changed

File: docs/development/code-conventions.rst [hunk 2]
```diff
@@ -516,27 +454,21 @@ allow you to debug the problem.
...
-operation. This is very handy when you want to run a set of tasks but
-are not sure what will happen when you do.
+Check mode allows Ansible to run your Playbooks in a dry-run sort of operation. This is handy when you want to run a set of tasks but are not sure what will happen when you do.
 
-Since BIG-IPs are usually considered a sensitive device to handle, there
-should always be a check-mode implemented in your module.
+Because BIG-IPs are usually considered a sensitive device to handle, you should always implement a check mode.
 
-.. _here: http://www.jeffgeerling.com/blog/yaml-best-practices-ansible-playbooks-tasks
+http://www.jeffgeerling.com/blog/yaml-best-practices-ansible-playbooks-tasks
```
Comment (caphrim007) [inline docs/development/code-conventions.rst]: this is the page that the ref was referring to. this needs to be reverted.

File: docs/development/code-conventions.rst [hunk 3]
```diff
@@ -564,14 +496,14 @@ use `delegate_to`.
          key: "setup.run"
          state: "reset"
 
-Default EXAMPLE parameters
-^^^^^^^^^^^^^^^^^^^^^^^^^^
+Format common parameters consistently
```
Comment (caphrim007) [inline docs/development/code-conventions.rst]: this header needs to be reverted. EXAMPLE is a literal block in the ansible module. So it is not sufficient to just say "format common parameters". We need to very specifically format the EXAMPLES parameter with the mentioned params.

Comment (selhorn) [reply docs/development/code-conventions.rst]: I am trying to make all the titles actions. Can I use something like "Set default EXAMPLE parameters"?

Comment (caphrim007) [reply docs/development/code-conventions.rst]: @selhorn that's fine

**Pull request merged**